tests/conftest.py

```diff
@@ -0,0 +1,76 @@
+import json
+import pytest
+import aioredis
+import asynctest
+
+import pyslackersweb
+import pyslackersweb.website.tasks
+
+pytest_plugins = ("slack.tests.plugin",)
+
+
+def pytest_addoption(parser):
+    parser.addoption("--redis", action="store_true", default=False)
+    parser.addoption("--postgresql", action="store_true", default=False)
+
+
+@pytest.fixture
+async def app(pytestconfig):
+    application = await pyslackersweb.app_factory()
+
+    if not pytestconfig.getoption("postgresql"):
+        application.cleanup_ctx.remove(pyslackersweb.contexts.postgresql_pool)
+
+    if not pytestconfig.getoption("redis"):
+        application.cleanup_ctx.remove(pyslackersweb.contexts.redis_pool)
+
+    return application
+
+
+@pytest.fixture
+async def redis(loop, pytestconfig):
+
+    if pytestconfig.getoption("redis"):
+        redis = await aioredis.create_redis_pool(pyslackersweb.settings.REDIS_URL)
+    else:
+        redis = FakeRedis()
+
+    yield redis
+
+    if pytestconfig.getoption("redis"):
+        redis.close()
+        await redis.wait_closed()
+
+
+class FakeRedis:
```

Comment:
I'm -1 on this, if we need to mock redis for tests, that's okay. But having a predefined class for the methods we may eventually use I am not super fond of. Personally I would rather just integration test for our needs and allow it to hit redis.